File: JUNGOL/Beginner/#1740 소수/소수.py
def main():
    MAX_NUM = 10000

    isPrime = [True for i in range(MAX_NUM+1)]
    isPrime[1] = False

    for i in range(4, MAX_NUM+1, 2): isPrime[i] = False
    for i in range(3, MAX_NUM+1, 2):
        if isPrime[i]:
            for j in range(i*2, MAX_NUM+1, i):
                isPrime[j] = False

    m = int(input())
    n = int(input())
    min_prime,sum = MAX_NUM,0
    for i in range(n, m-1, -1):
        if isPrime[i]:
            min_prime = i
            sum += i

    if sum == 0: print(-1)
    else: print(sum, min_prime, sep='\n')

main()

File: JUNGOL/Beginner/#1740 소수/소수.cpp
#include <iostream>
#include <vector>
using namespace std;

#define MAX_NUM 10000

int main() {
	// 에라토스테네스의 체
	vector<bool> isPrime(MAX_NUM+1, true);
	isPrime[1] = false;

	for(int i=4; i<=MAX_NUM; i+=2) isPrime[i] = false;
	for(int i=3; i<=MAX_NUM; i+=2){
		if(isPrime[i])
			for(int j=i*2; j<=MAX_NUM; j+=i)
				isPrime[j] = false;
	}

	int m,n; cin>>m>>n;
	int min_prime=MAX_NUM, sum=0;
	for(int i=n; i>=m; i--){
		if(isPrime[i]){
			min_prime = i;
			sum += i;
		}
	}

	if(sum == 0) printf("-1\n");
	else printf("%d\n%d\n", sum, min_prime);
}
